### ISSUE TITLE
fail to format javascript when inside an html file.

### ISSUE THREAD

**Issue by calidion** [Microsoft/vscode#20215]
- VSCode Version: Code 1.9.0 (27240e71ef390bf2d66307e677c2a333cebf75af, 2017-02-02T08:21:13.228Z)
- OS Version: Linux x64 4.8.0-37-generic
- Extensions:

|Extension|Author|Version|
|---|---|---|
|Angular2|UVBrain|0.3.5|
|html-snippets|abusaidm|0.0.18|
|project-manager|alefragnani|0.12.2|
|vscode-eslint|dbaeumer|1.2.2|
|vscode-stylesheet-formatter|dbalage|0.2.0|
|githistory|donjayamanne|0.1.4|
|python|donjayamanne|0.5.5|
|sass-lint|glen-84|0.0.3|
|docker-linter|henriiik|0.2.0|
|Angular2|johnpapa|1.0.2|
|Go|lukehoban|0.6.51|
|prettify-json|mohsen1|0.0.3|
|debugger-for-chrome|msjsdiag|2.4.1|
|sass-indented|robinbentley|1.3.0|
|addDocComments|stevencl|0.0.8|
|JavaScriptSnippets|xabikos|1.2.0|
|Material-theme|zhuangtongfa|1.0.3|

---

Steps to Reproduce:

1. for code

```
<script src="https://www.gstatic.com/firebasejs/3.6.9/firebase.js"></script>
<script>
  // Initialize Firebase
  var config = {
    apiKey: "A",
    authDomain: "ren",
    databaseURL: "https://renxin-a35b4.firebaseio.com",
    storageBucket: "ren",
    messagingSenderId: "54762"
  };
  firebase.initializeApp(config);
</script>
```

2. select format document

3. you will get

```

  <script src="https://www.gstatic.com/firebasejs/3.6.9/firebase.js"></script>
  <script>
    // Initialize Fireb    e
  var config           apiKey: "          authDomain: "re          databaseURL: "https://renxin-a35b4.firebaseio.co          storageBucket: "re          messagingSenderId: "547    "
     ;
  firebase.initializeApp(config);

  </script>
```

many chars are removed inside the script tag.

**Comment by mjbvz** [comment]
I believe this is a duplicate of  #19813

@aeschli Has a fix checked in to master and we plan on releasing a 1.9 recovery build with the fix as well

**Comment by aeschli** [comment]
Yes, the fix is in both master and 1.9, so the fix can be verified in the insiders build.
